spark/src/main/scala/geotrellis/spark/clip/ClipToGrid.scala

```diff
@@ -132,73 +140,53 @@ object ClipToGrid {
         def coveredBy(e: Extent) = pg.coveredBy(e.toPolygon.jtsGeom)
       }
 
-    def gcPredicates =
+    val gcPredicates =
       new Predicates {
         def covers(e: Extent) = feature.geom.jtsGeom.covers(e.toPolygon.jtsGeom)
         def coveredBy(e: Extent) = feature.geom.jtsGeom.coveredBy(e.toPolygon.jtsGeom)
       }
 
-    def clipToKey(
-      k: SpatialKey,
-      preds: Predicates
-    ): Option[(SpatialKey, Feature[Geometry, D])] = {
-      val extent = mapTransform(k)
-
-      clipFeature(extent, feature, preds).map(k -> _)
-    }
-
     val iterator: Iterator[(SpatialKey, Feature[Geometry, D])] =
       feature.geom match {
-        case p: Point =>
-          val k = mapTransform(p)
-          clipToKey(k, pointPredicates).toSeq.iterator
+        case p: Point => Iterator.single(mapTransform(p) -> feature)
```

Comment:
`Point` will never be outside its own `Extent`, so we can skip all attempts at clipping. The point predicates *do* help with skipping a lot of the work in `clipToKey`, but by avoiding that altogether we perform even fewer operations.

Reply:
This assumes that we don't want to change the metadata for the geometry in the clip function. That's an OK assumption.

Reply:
It also assumes something else: that user-invented clipping functions and `Predicates` will always be well-defined. We can imagine a case where the user is passing in clipping functions that do nonsensical things, such that a `Point` being outside of its Extent could be significant. This sounds like "garbage in, garbage out" though, and will probably never come up.

Which made me think that `Predicates` can probably be hidden from the user entirely. Is the set of `Predicates` you've provided hands-down the best? If so, we might be able to apply them in all cases and hide them from the API.